include/camp/resource.hpp

```diff
@@ -266,4 +311,26 @@ namespace resources
   }  // namespace v1
 }  // namespace resources
 }  // namespace camp
+
+/*
+ * \brief Specialization of std::hash for camp::resources::Resource.
+ * 
+ * Provides a hash function for Resource objects, enabling their use as keys
+ * in unordered associative containers (std::unordered_map, std::unordered_set, etc.).
+ * The hash is based on the Resource's unique identifier, which represents the
+ * combination of platform type and underlying stream/queue. Resources with 
+ * identical execution contexts (same platform and stream/queue) will produce
+ * the same hash value.
+ * 
+ * \return A size_t hash value computed from the Resource's internal ID.
+ */
+namespace std {
+  template <>
+  struct hash<camp::resources::Resource> {
```

Comment:
Can we add a hash specialization for each of the typed resources as well?

Reply:
Added (but not to the host)